server/api/src/main/java/org/hawkular/apm/server/api/task/ProcessingUnit.java

```diff
@@ -102,32 +104,40 @@ public void handle(String tenantId, List<T> items) throws Exception {
 
         processor.initialise(tenantId, items);
 
-        for (int i = 0; i < items.size(); i++) {
+        if (processor.getType() == ProcessorType.ManyToMany) {
             try {
-                if (processor.isMultiple()) {
-                    List<R> result = processor.processMultiple(tenantId, items.get(i));
-                    if (resultHandler != null && result != null && !result.isEmpty()) {
-                        if (results == null) {
-                            results = new ArrayList<R>();
+                results = processor.processManyToMany(tenantId, items);
+            } catch (Exception e) {
+                retries = items;
+            }
+        } else {
+            for (int i = 0; i < items.size(); i++) {
```

Comment:
Wouldn't it be better to use a `stream` here, instead of a loop? Due to the operations done inside the loop, I think this is a perfect candidate for streams :)